crates/sound/src/vhu_sound.rs

```diff
@@ -1,23 +1,40 @@
 // SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
 
-use crate::audio_backends::{alloc_audio_backend, AudioBackend};
-use crate::virtio_sound::*;
-use crate::{Error, Result, SoundConfig};
-
+use std::mem::size_of;
 use std::sync::RwLock;
 use std::{io::Result as IoResult, u16, u32, u64, u8};
+
+use log::{error, debug};
 use vhost::vhost_user::message::{VhostUserProtocolFeatures, VhostUserVirtioFeatures};
-use vhost_user_backend::{VhostUserBackend, VringRwLock};
+use vhost_user_backend::{VhostUserBackend, VringRwLock, VringT};
 use virtio_bindings::bindings::{
     virtio_config::VIRTIO_F_NOTIFY_ON_EMPTY, virtio_config::VIRTIO_F_VERSION_1,
     virtio_ring::VIRTIO_RING_F_EVENT_IDX, virtio_ring::VIRTIO_RING_F_INDIRECT_DESC,
 };
-use vm_memory::{ByteValued, GuestMemoryAtomic, GuestMemoryMmap};
+use virtio_queue::{DescriptorChain, QueueOwnedT};
+use vm_memory::{Bytes, ByteValued, GuestAddressSpace, GuestMemoryAtomic, GuestMemoryMmap, GuestMemoryLoadGuard};
 use vmm_sys_util::{
     epoll::EventSet,
     eventfd::{EventFd, EFD_NONBLOCK},
 };
 
+use crate::audio_backends::{alloc_audio_backend, AudioBackend};
+use crate::virtio_sound::*;
+use crate::{Error, Result, SoundConfig};
+
+pub const SUPPORTED_FORMATS: u64 = 1 << VIRTIO_SND_PCM_FMT_U8
```

Comment:
It seems `SUPPORTED_FORMATS` and `SUPPORTED_RATES` are used only in this module, so we can remove `pub`